### ISSUE TITLE
`SKPaint.MeasureText` returns incorrect value

### ISSUE THREAD

**Issue by tdenniston** [mono/SkiaSharp#78]
It looks like the `MeasureText` method is computing the length of strings incorrectly. Here is an example of how the incorrect bounds are computed for text. I'm using the following snippet:

```
paint.Color = SKColors.Black;
paint.IsAntialias = true;
paint.TextEncoding = SKTextEncoding.Utf16;
paint.TextSize = 45;
SKRect bounds = new SKRect();
paint.MeasureText(text, ref bounds);
```

And the `bounds` rect is rendered like this, for several strings of differing lengths:
![text-width-bug](https://cloud.githubusercontent.com/assets/934815/15088086/6989e762-13bd-11e6-8cd5-74c819a4da31.png)

When I pad the string with spaces like so:

```
paint.Color = SKColors.Black;
paint.IsAntialias = true;
paint.TextEncoding = SKTextEncoding.Utf16;
paint.TextSize = 45;
SKRect bounds = new SKRect();
string padded = text;
for (int i = 0; i < text.Length; i++) padded += " ";
paint.MeasureText(padded, ref bounds);
```

The bounds are computed correctly:

![text-width-bug-padded](https://cloud.githubusercontent.com/assets/934815/15088139/da142452-13bd-11e6-9d74-05062ad499eb.png)

**Comment by mattleibow** [comment]
@tdenniston Thanks for reporting this. Although it was fixed a little while ago, I re-fixed it.

Basically, the error was that we were assuming UTF-8 (which was totally wrong since we could have UTF-16 and UTF-32 in the paint object). Then it was fixed to force UTF-16, although this "worked" we changed the paint's encoding to UTF-16 for a few milliseconds... and "forcing" is not always good. This now uses the encoding of the paint.

To draw UTF-32 text, the paint object must be ready to do so, thus you must set the `TextEncoding` property to `SKTextEncoding.Utf32`. In fact, this is what that property is for:

> Describes how to interpret the text parameters that are passed to paint methods like `MeasureText()` and `GetTextWidths()`.
